### PR TITLE
Updating provider_cldr and testdata to CLDR 38.1.0-BETA4

### LINKED ISSUE #409: Upgrade to CLDR 38
@srl295 has done work to improve the structure CLDR 38 JSON.  We will need to change ICU4X to use the new JSON distribution mechanism before we can upgrade to CLDR 38.

CC @zbraniecki @filmil

### PR DESCRIPTION
**Pull request by sffc**
Fixes #409
Depends on #407

I implemented a downloader for the new all-in-one CLDR JSON distribution.  I can't switch the testdata over to CLDR 38, though, until we support the "e" operand.  Right now I get the error:

> thread 'main' panicked at 'Rule parsing failed.: ExpectedOperand', components/provider_cldr/src/transform/plurals.rs:141:47

### CODE REVIEW

File: components/provider_cldr/src/download/cldr_components.rs [hunk 1]
```diff
@@ -1,6 +1,10 @@
 // This file is part of ICU4X. For terms of use, please see the file
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+
+//! Data downloader for CLDR JSON versions 37 and older. Prior to CLDR 38, the JSON for each
```
Comment (sffc) [inline components/provider_cldr/src/download/cldr_components.rs:5]: I could have just deleted this code.  I don't like introducing legacy code so early in this project's lifetime.  Thoughts?

Comment (srl295) [reply components/provider_cldr/src/download/cldr_components.rs:5]: my thought is that deleting the code would prevent comparison w/ CLDR 37, would be harder to track down issues.

**Review by srl295 — APPROVED** [review]
LGTM

File: components/provider_cldr/src/download/cldr_components.rs (hunk 1 above)
Comment (zbraniecki) [reply components/provider_cldr/src/download/cldr_components.rs:5]: I'd remove the code. It's way to early for us to keep legacy code for any user who might want to rely on it. We should stay light on code and nimble as possible. Adding more legacy code to maintain in provider now is the opposite for me.

Comment (srl295) [reply components/provider_cldr/src/download/cldr_components.rs:5]: @zbraniecki its' a small change, and easy enough for someone to pull out of the archives if need be.

No objection to deletion.

Comment (sffc) [reply components/provider_cldr/src/download/cldr_components.rs:5]: How hard would it be to release old CLDR versions using the new zip file format, at least CLDR 36 and 37?

**Review by zbraniecki — APPROVED** [review]

File: components/provider_cldr/src/download/cldr_components.rs (hunk 1 above)
Comment (sffc) [reply components/provider_cldr/src/download/cldr_components.rs:5]: I removed the CLDR 37 support now that ICU4X supports CLDR 38.

**Review by zbraniecki — APPROVED** [review]
lgtm!

**Pull request merged**